File: New_socket_proj/scratch_regex.js

const raw = "\nIP                 PORT     LOAD (%)     ACTIVE      \n────────────────   ──────   ───────────  ─────────── \n172.30.2.21        7777     24.2%        0           \n";

const re = /(\d+\.\d+\.\d+\.\d+)[\s:]+(\d+)\s+([\d.]+)\s*%?\s+(\d+)?/g;
let m;
console.log("Testing regex...");
while ((m = re.exec(raw)) !== null) {
  console.log("Match found!");
  console.log("IP:", m[1]);
  console.log("Port:", m[2]);
  console.log("Load:", m[3]);
  console.log("Tasks:", m[4]);
}
if (!raw.match(re)) {
  console.log("No matches found.");
}

File: New_socket_proj/scratch_regex.py

import re

raw = "\nIP                 PORT     LOAD (%)     ACTIVE      \n────────────────   ──────   ───────────  ─────────── \n172.30.2.21        7777     24.2%        0           \n"

# The regex from app.js: (\d+\.\d+\.\d+\.\d+)[\s:]+(\d+)\s+([\d.]+)\s*%?\s+(\d+)?
pattern = r"(\d+\.\d+\.\d+\.\d+)[\s:]+(\d+)\s+([\d.]+)\s*%?\s+(\d+)?"

matches = list(re.finditer(pattern, raw))
print(f"Testing regex... Found {len(matches)} matches.")

for m in matches:
    print(f"Match found!")
    print(f"IP: {m.group(1)}")
    print(f"Port: {m.group(2)}")
    print(f"Load: {m.group(3)}")
    print(f"Tasks: {m.group(4)}")
